store/copr/coprocessor.go

```diff
@@ -1255,11 +1342,20 @@ func (worker *copIteratorWorker) handleBatchCopResponse(bo *Backoffer, batchResp
 		return nil, nil
 	}
 	var remainTasks []*copTask
+	appendRemainTasks := func(tasks ...*copTask) {
+		if remainTasks == nil {
+			// allocate size fo remain length
+			remainTasks = make([]*copTask, 0, len(tasks))
+		}
+		remainTasks = append(remainTasks, tasks...)
+	}
 	for _, batchResp := range batchResps {
-		batchedTask, ok := tasks[batchResp.GetTaskId()]
+		taskID := batchResp.GetTaskId()
+		batchedTask, ok := tasks[taskID]
 		if !ok {
 			return nil, errors.Errorf("task id %d not found", batchResp.GetTaskId())
 		}
+		delete(tasks, taskID)
```

Comment:
This may change the input `tasks` and make it a mutable reference, better to comment about it in the function declaration.